test/unit/assorted/server_discovery_and_monitoring.spec.test.ts

```diff
@@ -322,84 +326,88 @@ async function executeSDAMTest(testData: SDAMTest) {
   // connect the topology
   await client.connect();
 
-  for (const phase of testData.phases) {
-    // Determine which of the two kinds of phases we're running
-    if ('responses' in phase && phase.responses != null) {
-      // phase with responses for hello simulations
-      for (const [address, hello] of phase.responses) {
-        client.topology.serverUpdateHandler(new ServerDescription(address, hello));
-      }
-    } else if ('applicationErrors' in phase && phase.applicationErrors) {
-      // phase with applicationErrors simulating error's from network, timeouts, server
-      for (const appError of phase.applicationErrors) {
-        // Stub will return appError to SDAM machinery
-        const checkOutStub = sinon
-          .stub(ConnectionPool.prototype, 'checkOut')
-          .callsFake(checkoutStubImpl(appError));
-
-        const server = client.topology.s.servers.get(appError.address);
-
-        // Run a dummy command to encounter the error
-        const res = server.command.bind(server)(ns('admin.$cmd'), { ping: 1 }, {});
-        const thrownError = await res.catch(error => error);
-
-        // Restore the stub before asserting anything in case of errors
-        checkOutStub.restore();
-
-        const isApplicationError = error => {
-          // These errors all come from the withConnection stub
-          return (
-            error instanceof MongoNetworkError ||
-            error instanceof MongoNetworkTimeoutError ||
-            error instanceof MongoServerError
-          );
-        };
-        expect(
-          thrownError,
-          `expected the error thrown to be one of MongoNetworkError, MongoNetworkTimeoutError or MongoServerError (referred to in the spec as an "Application Error") got ${thrownError.name} ${thrownError.stack}`
-        ).to.satisfy(isApplicationError);
+  try {
```

Comment:
How were these tests not leaking clients before this change? 🤔

Reply:
Not entirely sure how they weren't leaking in CI, but when I ran them locally they were 🤷🏿

Reply:
🤔 

Well, nice catch!

Reply:
There's no leak checker on unit tests (per the .mocharc.json config) because unit tests shouldn't create any real connections. Fine with the change because it is logically "balanced" but was there actually a hang or some issue that required this?

Reply:
When testing locally, mocha was actually complaining about open mongo clients